### PR TITLE
PUBDEV-5254 - GLM Multinomial is not working properly.

### PR DESCRIPTION
**Pull request by Pscheidl**

### CODE REVIEW

**Comment by Pscheidl** [comment]
This PR is a fix for approved, yet badly merged https://github.com/h2oai/h2o-3/pull/2003. Tests are fixed in addition.

**Comment by Pscheidl** [comment]
No test code has been commented out, that was the previous state. Looks like naive bayes has been part of some tests for a long ago.

The "unsupported" state of naive bayes should be reverted ?

**Comment by wendycwong** [comment]
@michalKurka:

For the multinomial, I think it is okay not to worry about the naive coordinate descent tests.  The multinomial link function is not implemented.  I can see the difficulty of implementing the link function because it is exp(class i)/(sum over all classes of exp(classk).  Since the link function only takes one argument, it cannot do it for multinomial.

I have an alternative suggestion.  Multinomial has not been tested against other multinomial implement in R or Python.  Hence, I will like to see tests of multinomial results against R and Python toolboxes from third party. 

A good starting point can the the pyunit_PUBDEV_5008_glm_ordinal_large.py.  This test is derived from the gridsearch test which tests many aspects of the algo.  Let me modified it for my ordinal regression test and it can be used for multinomial test later.

**Comment by Pscheidl** [comment]
WIP. There are two tests failing that are not failing locally. Not related to this PR.

However, I am still not satisfied with the R test.

**Comment by Pscheidl** [comment]
The R test has been improved to my satisfaction - there is an assert for GLM using naive descent should not pass. When I return back in future to reimplement this part of GLM, it will be forced to fix this test and thus have a look at it and consider its validity.

File: h2o-algos/src/main/java/hex/glm/GLMModel.java [hunk 1]
```diff
@@ -211,6 +212,8 @@ public GLMModel clone2(){
     public boolean _stdOverride; // standardization override by beta constraints
 
     public void validate(GLM glm) {
+      if (_solver.equals(Solver.COORDINATE_DESCENT_NAIVE))
+        throw H2O.unimpl("Naive coordinate descent is not supported.");
```
Comment (wendycwong) [inline h2o-algos/src/main/java/hex/glm/GLMModel.java]: Is COORDINATE_DESCENT_NAIVE not supported for multinomial?  Or for families as well?

Comment (Pscheidl) [reply h2o-algos/src/main/java/hex/glm/GLMModel.java]: This has been removed in the next commit.

**Comment by michalkurka** [comment]
Test Result (4 failures / +3)
		Java 8 JUnit / hex.glm.GLMBasicTestRegression.testWeights
		Java 8 JUnit / hex.glm.GLMTest.testCoordinateDescent_airlines
		Java 8 JUnit / hex.glm.GLMTest.testCoordinateDescent_anomaly
		Java 8 JUnit / hex.glm.GLMTest.test_COD_Airlines_SingleLambda

**Comment by Pscheidl** [comment]
Strange. These tests work locally. And worked in the previous iteration. Let's just build again.

**Comment by michalkurka** [comment]
@Pscheidl seems ready to be merged

**Pull request merged**